### PR TITLE
Fix problematic to_arrow tests

### LINKED ISSUE #2514: Python API test_arrow.test is not enabled
After renabling, test_to_arrow_complex and test_to_arrow1 fails

### PR DESCRIPTION
**Pull request by mxwli**
Adds `UNION` exporting to `get_as_arrow`
Adds `MAP` exporting to `get_as_arrow`
Fixes `REL` exporting to `get_as_arrow`
Fixes `INTERVAL` exporting to `get_as_arrow`
Fixes broken tests `_test_node`, `_test_node_rel`, `test_marries_table` in `test_arrow.py::test_to_arrow_complex`
Does not fix `test_to_arrow1`, which calls `get_as_arrow(-1)` (what even is the expected behaviour of this?)

Mostly closes #2514 

Adds coverage to #2539

### CODE REVIEW

**Review by andyfengHKU — APPROVED** [review]
LGTM overall. Maybe let @ray6080 take another look tonight?

File: Makefile [hunk 1]
```diff
@@ -111,7 +111,7 @@ nodejs:
 	$(call run-cmake-release, -DBUILD_NODEJS=TRUE)
 
 python:
-	$(call run-cmake-release, -DBUILD_PYTHON=TRUE)
+	$(call run-cmake-debug, -DBUILD_PYTHON=TRUE)
```
[the remark this answers is not in the document]
Comment (andyfengHKU) [reply Makefile:114]: sure

File: src/common/arrow/arrow_row_batch.cpp [hunk 2]
```diff
@@ -76,6 +76,21 @@ void ArrowRowBatch::templateInitializeVector<LogicalTypeID::STRUCT>(ArrowVector*
     initializeStructVector(vector, type, capacity);
 }
 
+template<>
+void ArrowRowBatch::templateInitializeVector<LogicalTypeID::UNION>(ArrowVector* vector,
+    const LogicalType& type, std::int64_t capacity) {
+    // Interestingly, unions don't have their own validity bitmap
```
[the remark this answers is not in the document]
Comment (mxwli) [reply src/common/arrow/arrow_row_batch.cpp:82]: Yes, Arrow Union Arrays refer to their children for validity bitmaps.

File: src/common/arrow/arrow_row_batch.cpp [hunk 3]
```diff
@@ -187,6 +202,9 @@ std::unique_ptr<ArrowVector> ArrowRowBatch::createVector(const LogicalType& type
     case LogicalTypeID::STRUCT: {
         templateInitializeVector<LogicalTypeID::STRUCT>(result.get(), type, capacity);
     } break;
+    case LogicalTypeID::UNION: {
```
[the remark this answers is not in the document]
Comment (mxwli) [reply src/common/arrow/arrow_row_batch.cpp:205]: That seems to be the case, yes. I can add that in this PR as well; MAPs are physically LISTs in arrow.

File: src/common/arrow/arrow_converter.cpp [hunk 4]
```diff
@@ -76,10 +99,10 @@ void ArrowConverter::setArrowFormatForInternalID(ArrowSchemaHolder& rootHolder,
     }
     child.children = &rootHolder.nestedChildrenPtr.back()[0];
     initializeChild(*child.children[0]);
-    child.children[0]->name = copyName(rootHolder, "table");
+    child.children[0]->name = copyName(rootHolder, "offset");
     setArrowFormat(rootHolder, *child.children[0], *LogicalType::INT64());
     initializeChild(*child.children[1]);
-    child.children[1]->name = copyName(rootHolder, "offset");
+    child.children[1]->name = copyName(rootHolder, "table");
```
Comment (ray6080) [inline src/common/arrow/arrow_converter.cpp:105]: Was this a bug? I saw you changed the order of `offset` and `table` here.

File: src/common/arrow/arrow_row_batch.cpp [hunk 5]
```diff
@@ -70,12 +70,33 @@ void ArrowRowBatch::templateInitializeVector<LogicalTypeID::ARRAY>(ArrowVector*
...
 template<>
 void ArrowRowBatch::templateInitializeVector<LogicalTypeID::STRUCT>(ArrowVector* vector,
     const LogicalType& type, std::int64_t capacity) {
     initializeStructVector(vector, type, capacity);
 }
 
+template<>
+void ArrowRowBatch::templateInitializeVector<LogicalTypeID::UNION>(ArrowVector* vector,
+    const LogicalType& type, std::int64_t capacity) {
+    // Interestingly, unions don't have their own validity bitmap
```
Comment (ray6080) [inline src/common/arrow/arrow_row_batch.cpp:88]: Are u aware why is that? maybe put a link to arrow's doc here? so the comment is more self-explanatory.

**Review by ray6080 — APPROVED** [review]

File: src/common/arrow/arrow_converter.cpp (hunk 4 above)
Comment (mxwli) [reply src/common/arrow/arrow_converter.cpp:105]: It turns out it was in the wrong order initially

File: src/common/arrow/arrow_row_batch.cpp (hunk 5 above)
Comment (mxwli) [reply src/common/arrow/arrow_row_batch.cpp:88]: Added link to doc :+1:

**Pull request merged**